alf/utils/lean_function.py

```diff
@@ -13,67 +13,123 @@
 # limitations under the License.
 
 import torch
+import torch.nn as nn
+import types
+from typing import Callable
+
+from alf.nest import flatten, pack_sequence_as
+from alf.networks import Network
 
 
 class _LeanFunction(torch.autograd.Function):
+    # Reference: Pytorch: Defining new Autograd Functions
+    # https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html
     @staticmethod
-    def forward(ctx, func, keywords, *args):
+    def forward(ctx, func, num_parameters, keywords, *args):
+        """
+        Args:
+            ctx (_LeanFunction):
```

Comment:
Is is left blank intentionally? I think it is still worth it to explain this context variable even if it might be trivial for `autograd.Function`.

Reply:
added.